codice/alliance
# CAL-204: Convert NITF plugin to be asynchronous PostProcessPlugin

#### What does this PR do?
_This PR requires ~~[DDF-2569](https://github.com/codice/ddf/pull/1495)~~, ~~[IMG-206](https://github.com/codice/imaging-nitf/pull/145)~~, and ~~[DDF-2871](https://github.com/codice/ddf/pull/1792)~~_

~~DO NOT MERGE until https://codice.atlassian.net/browse/CAL-286~~

- Converts the `NitfPreStoragePlugin` to be an asynchronous `PostProcessPlugin`.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@bdeining @emmberk @troymohl @dcruver @glenhein 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jaymcnallie
@jlcsmith

#### How should this be tested?
_See me if you need some NITFs_
- Full build with itests passing
- Ensure large NITFs no longer take minutes to ingest.
- Ingest a NITF and search for it. There should be no thumbnail, overview, or original image on the metacard before the update comes in.
- Once the update comes in, search again and ensure the thumbnail, overview, and original image are all present. Ensure the product can be downloaded. Can use `log:set TRACE org.codice.ddf.catalog.async` to see logs.
- Test that chipping still works (see https://github.com/codice/alliance/pull/216)
- Test different types of nitfs (including those with bands and jpeg200)
- Confirm that changing the `NITF Render Plugin` configuration still works as expected
- Ensure `data/tmp` and `content/store/tmp` does not have any left over resources. Especially when ingesting a large amount of files at a time.

#### Any background context you want to provide?
There was a need for NITF image processing to be asynchronous due to large files taking large processing times.

#### What are the relevant tickets?
[CAL-204](https://codice.atlassian.net/browse/CAL-204)

#### Screenshots (if appropriate)
#### Checklist:
